python/ray/exceptions.py

```diff
@@ -292,19 +295,103 @@ def __init__(self, object_ref_hex, call_site):
         if self.call_site:
-            msg += (f" The ObjectRef was created at: {self.call_site}")
+            msg += (f"The ObjectRef was created at: {self.call_site}")
         else:
             msg += (
-                " To see information about where this ObjectRef was created "
+                "To see information about where this ObjectRef was created "
                 "in Python, set the environment variable "
                 "RAY_record_ref_creation_sites=1 during `ray start` and "
                 "`ray.init()`.")
         return msg
 
 
+class ObjectLostError(ObjectUnreachableError):
+    """Indicates that the object is lost from distributed memory, due to
+    node failure or system error.
+
+    Attributes:
+        object_ref_hex: Hex ID of the object.
+    """
+
+    def __str__(self):
+        return super().__str__() + "\n\n" + (
+            f"All copies of {self.object_ref_hex} are lost "
+            "due to node failure.\n\n"
+            "If you did not receive a message about a worker node "
+            "dying, this is likely a system-level bug. "
+            "Please file an issue on GitHub at "
+            "https://github.com/ray-project/ray/issues/new/choose.")
+
+
+class ObjectReleasedError(ObjectUnreachableError):
+    """Indicates that an object has been released while there was still a
+    reference to it.
+
+    Attributes:
+        object_ref_hex: Hex ID of the object.
+    """
+
+    def __str__(self):
+        return super().__str__() + "\n\n" + (
+            f"Object {self.object_ref_hex} has already been released.\n\n"
+            "This is likely due to a corner case in the distributed "
+            "reference counting protocol that can occur when a worker passes "
+            "an ObjectRef, then exits before the ref count at the "
+            "ObjectRef's owner can be updated. For example, suppose we "
+            "call x_ref = foo.remote(...), pass [x_ref] to an actor A, and "
+            "then actor A passes x_ref to actor B by calling "
+            "B.bar.remote(x_ref). In this case, the driver may release x "
+            "before B.bar executes, and B will receive this error.\n\n"
+            "If your application does not match this scenario, then this is "
+            "likely a system-level bug in the distributed ref counting "
+            "protocol. Please file an issue on GitHub at "
+            "https://github.com/ray-project/ray/issues/new/choose.")
+
+
+class OwnerDiedError(ObjectUnreachableError):
+    """Indicates that the owner of the object has died while there is still a
+    reference to the object.
+
+    Attributes:
+        object_ref_hex: Hex ID of the object.
+    """
+
+    def __str__(self):
+        return super().__str__() + "\n\n" + (
+            f"Object {self.object_ref_hex} cannot be retrieved because "
+            "the Python worker that first created the ObjectRef (via "
+            "`.remote()` or `ray.put()`) has exited. "
```

Comment:
`Python worker that first created the ObjectRef (owner) via ".remote"...` so that users would understand the meaning of Owner (since it is mentioned in the Exception name)